ProductDetailScreen.tsx

```diff
@@ -0,0 +1,153 @@
+import React, { useState, useEffect } from "react";
+import {View,Text,TouchableOpacity,Image,StyleSheet,ActivityIndicator,} from "react-native";
+import AsyncStorage from "@react-native-async-storage/async-storage";
+import { useRoute } from "@react-navigation/native";
+import { useFetch } from "./useFetch";
+
+const ProductDetailScreen = () => {
+  const [isFavorite, setIsFavorite] = useState(false);
+  const route = useRoute();
+  const { url } = route.params;
+
+  const { productData, loading, error } = useFetch(url);
+
+  const toggleFavorite = async () => {
+    const newFavoriteStatus = !isFavorite;
+    setIsFavorite(newFavoriteStatus);
+    await AsyncStorage.setItem(url.toString(), newFavoriteStatus.toString());
```

Comment:
One of the hints I provided was: "You should save your favorite products as a single array of objects in async storage. This way, you only have to worry about one key." 

I'm not taking any points away for this, but managing this in a single array is one of the challenges I wanted students to solve. There's no easy way to retrieve these keys unless you use `getAllKeys` which will get _everything_ out of async storage. In a production application, you could have a bunch of different things stored, and you wouldn't want to fetch everything each time you wanted to grab a single item from storage.